maestro-client/src/main/java/maestro/drivers/AndroidDriver.kt

```diff
@@ -519,26 +519,60 @@ class AndroidDriver(
     }
 
     override fun setPermissions(appId: String, permissions: Map<String, String>) {
-        val mutable = permissions.toMutableMap()
-        mutable.remove("all")?.let { value ->
-            setAllPermissions(appId, value)
-        }
-
-        mutable.forEach { permission ->
-            setPermission(appId, permission.key, permission.value)
+        permissions.forEach { (permission, permissionValue) ->
+            val translatedPermissionValue = translatePermissionValue(permissionValue)
+            when(permission.replace("[^A-Za-z0-9._]+".toRegex(), "")) {
+                "all" -> setAllPermissions(appId, translatedPermissionValue)
```

Comment:
This isn't deterministic. The permissions are a hash and can be in any order. When a permission is set like:

```
- launchApp: { permissions: { all: allow, notifications: deny } }
```

It depends on the runtime ordering of the hash elements weather notifications will be set to allow or deny.
Please consider reverting to what was there:

```
val mutable = permissions.toMutableMap()
         mutable.remove("all")?.let { value ->
             setAllPermissions(appId, value)
         }

         mutable.forEach { permission ->
             setPermission(appId, translatePermissionName(permission.key),  translatePermissionValue(permission.value))
         }
```